lib/statsd/instrument/datagram_builder.rb

```diff
@@ -104,30 +104,43 @@ def generate_generic_datagram(name, value, type, sample_rate, tags)
       end
 
       def compile_tags(tags, buffer = "".b)
+        @tag_cache ||= {}
```

Comment:
note that the cache is only scoped to the individual datagram_builder instance, so if a new instance is used we won't benefit from memoizing.